packages/core/src/amazonq/webview/ui/quickActions/handler.ts

```diff
@@ -343,6 +325,33 @@ export class QuickActionHandler {
         }
     }
 
+    private addTab(affectedTabId: string | undefined) {
+        if (!affectedTabId || !this.mynahUI) {
+            return
+        }
+
+        const currTab = this.mynahUI.getAllTabs()[affectedTabId]
+        const currTabWasUsed =
+            (currTab.store?.chatItems?.filter((item) => item.type === ChatItemType.PROMPT).length ?? 0) > 0
+        if (currTabWasUsed) {
```

Comment:
what exactly is this case for? What do we do differently when the tab is used/not used?

Reply:
It was a note from Oksana's original p.o.c: https://github.com/aws/language-servers/pull/733/files. As far as I understand in flare you can't create a tab and then later convert it. In the old version we were basing things on tab types which wouldn't exist in chat-client so instead we're just checking if someones interacted with the current tab before re-using it. If its used we create a new tab, otherwise we re-use the current tab